File: samples/java/app/src/main/java/org/remdev/wetweexapp/generator/GeneratorActivity.java
package org.remdev.wetweexapp.generator;

import android.os.Bundle;
import android.widget.EditText;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import by.nalivajr.tweexdemo.R;
import org.remdev.wetweex.activity.BaseTweexProgressObservingActivity;
import org.remdev.wetweex.component.CoordinatorsContainer;
import org.remdev.wetweexapp.DemoAppCoordinatorsContainer;
import org.remdev.wetweexapp.Dependencies;
import org.remdev.wetweexapp.Strategy;

public class GeneratorActivity extends BaseTweexProgressObservingActivity<GeneratorViewModel, GeneratorCoordinator> {

    private TextView text;
    private TextView generateCancellable;
    private TextView generateCrucial;
    private TextView generateNonCancellable;
    private TextView cancel;
    private EditText generationTime;


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_generator);

        text = findViewById(R.id.text);
        generateCancellable = findViewById(R.id.generate_button_c);
        generateCrucial = findViewById(R.id.generate_button_cr);
        generateNonCancellable = findViewById(R.id.generate_button_nc);
        cancel = findViewById(R.id.cancel_button);
        generationTime = findViewById(R.id.generation_time);

        getViewModel().getGeneratedText().observe(this, text::setText);
        generateCancellable.setOnClickListener(__ -> startGenertion(Strategy.CANCELLABLE));
        generateNonCancellable.setOnClickListener(__ -> startGenertion(Strategy.NON_CANCELLABLE));
        generateCrucial.setOnClickListener(__ -> startGenertion(Strategy.CRUCIAL));
        cancel.setOnClickListener(__ -> getCoordinator().tryCancelAll());

    }

    private void startGenertion(Strategy strategy) {
        long millis = Long.parseLong(generationTime.getText().toString()) * 1000L;
        getCoordinator().generateString(strategy, millis);
    }

    @NonNull
    @Override
    protected GeneratorViewModel createViewModel() {
        return new GeneratorViewModel();
    }

    @NonNull
    @Override
    protected GeneratorCoordinator createCoordinator(@NonNull GeneratorViewModel generatorViewModel) {
        return new GeneratorCoordinator(generatorViewModel, Dependencies.textGenerator());
    }

    @NonNull
    @Override
    protected CoordinatorsContainer getCoordinatorContainer() {
        return DemoAppCoordinatorsContainer.getInstance();
    }
}

File: samples/kotlin/app/src/main/java/org/remdev/wetweexapp/generator/GeneratorActivity.kt
package org.remdev.wetweexapp.generator

import android.os.Bundle
import android.widget.EditText
import android.widget.TextView
import androidx.lifecycle.Observer
import org.remdev.wetweex.activity.BaseTweexProgressObservingActivity
import org.remdev.wetweex.component.CoordinatorsContainer
import org.remdev.wetweexapp.DemoAppCoordinatorsContainer
import by.nalivajr.tweexdemo.R
import org.remdev.wetweexapp.Strategy
import org.remdev.wetweexapp.textGenerator

class GeneratorActivity : BaseTweexProgressObservingActivity<GeneratorViewModel, GeneratorCoordinator>() {

    private val text by lazy { findViewById<TextView>(R.id.text) }
    private val generateCancellable by lazy { findViewById<TextView>(R.id.generate_button_c) }
    private val generateCrucial by lazy { findViewById<TextView>(R.id.generate_button_cr) }
    private val generateNonCancellable by lazy { findViewById<TextView>(R.id.generate_button_nc) }
    private val cancel by lazy { findViewById<TextView>(R.id.cancel_button) }
    private val generationTime by lazy { findViewById<EditText>(R.id.generation_time) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_generator)
        viewModel.generatedText.observe(this, Observer<String> {
            text.text = it
        })
        generateCancellable.setOnClickListener {
            startGenertion(Strategy.CANCELLABLE)
        }
        generateNonCancellable.setOnClickListener {
            startGenertion(Strategy.NON_CANCELLABLE)
        }
        generateCrucial.setOnClickListener {
            startGenertion(Strategy.CRUCIAL)
        }
        cancel.setOnClickListener {
            coordinator.tryCancelAll()
        }
    }

    private fun startGenertion(strategy: Strategy) {
        val millis = generationTime.text.toString().toLong() * 1000
        coordinator.generateString(strategy, millis)
    }

    override fun createViewModel(): GeneratorViewModel =
        GeneratorViewModel()

    override fun getCoordinatorContainer(): CoordinatorsContainer = DemoAppCoordinatorsContainer

    override fun createCoordinator(viewModel: GeneratorViewModel): GeneratorCoordinator
            = GeneratorCoordinator(viewModel, textGenerator)
}
